sdk/search/Azure.Search.Documents/tests/SearchIndexerClientTests.cs

```diff
@@ -486,6 +486,72 @@ public async Task CrudSkillset()
+                Source = "/document/tableprojection/pages/*/keyphrase/*",
+                SourceContext = null,
+            };
+
+            SearchIndexerKnowledgeStoreProjection projection1 = new();
+            projection1.Tables.Add(table1);
+            projection1.Tables.Add(table2);
+            projection1.Tables.Add(table3);
+
+            SearchIndexerKnowledgeStoreTableProjectionSelector table4 = new("hotelReviewsInlineDocument")
+            {
+                GeneratedKeyName = "Documentid",
+                Source = null,
+                SourceContext = "/document",
+            };
+            table4.Inputs.Add(new("name") { Source = "/document/name", SourceContext = null });
+            table4.Inputs.Add(new("reviews_date") { Source = "/document/reviews_date", SourceContext = null });
+            table4.Inputs.Add(new("reviews_rating") { Source = "/document/reviews_rating", SourceContext = null });
+            table4.Inputs.Add(new("reviews_text") { Source = "/document/reviews_text", SourceContext = null });
+            table4.Inputs.Add(new("reviews_title") { Source = "/document/reviews_title", SourceContext = null });
+            table4.Inputs.Add(new("AzureSearch_DocumentKey") { Source = "/document/AzureSearch_DocumentKey", SourceContext = null });
+
+            SearchIndexerKnowledgeStoreTableProjectionSelector table5 = new("hotelReviewsInlinePages")
+            {
+                GeneratedKeyName = "Pagesid",
+                Source = null,
+                SourceContext = "/document/reviews_text/pages/*",
+            };
+            table5.Inputs.Add(new("SentimentScore") { Source = "/document/reviews_text/pages/*/Sentiment", SourceContext = null });
+            table5.Inputs.Add(new("LanguageCode") { Source = "/document/Language", SourceContext = null });
+            table5.Inputs.Add(new("Page") { Source = "/document/reviews_text/pages/*", SourceContext = null });
+
+            SearchIndexerKnowledgeStoreTableProjectionSelector table6 = new("hotelReviewsInlineKeyPhrases")
+            {
+                GeneratedKeyName = "kpidv2",
+                Source = null,
+                SourceContext = "/document/reviews_text/pages/*/Keyphrases/*",
+            };
+            table6.Inputs.Add(new("Keyphrases") { Source = "/document/reviews_text/pages/*/Keyphrases/*", SourceContext = null });
+
+            SearchIndexerKnowledgeStoreProjection projection2 = new();
+            projection2.Tables.Add(table4);
+            projection2.Tables.Add(table5);
+            projection2.Tables.Add(table6);
+
+            List<SearchIndexerKnowledgeStoreProjection> projections = new() { projection1, projection2 };
+
+            skillset.KnowledgeStore = new SearchIndexerKnowledgeStore(resources.StorageAccountConnectionString, projections);
+
             // Create the skillset.
             SearchIndexerSkillset createdSkillset = await client.CreateSkillsetAsync(skillset);
```

Comment:
I am looking into a failure in this test @tg-msft. If you can provide any hints, that would be great.
In `CrudSkillset()` test, The POST call for `CreateSkillsetAsync()` on line 556 is succeeding, but the PUT call to `CreateOrUpdateSkillsetAsync()` on line 564 is failing with this message -
_Unable to find a record for the request PUT https://mohitc-acs.search.windows.net/skillsets('qgrsmesk')?api-version=2020-06-30-Preview_

Looking at CrudSkillset[Async].json, I can see that the PUT request has a 200 response with the data.

Reply:
What does the full output say?  I think it usually [gives some hints](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core.TestFramework/src/RecordMatcher.cs#L201) at how your record differs from the best guess.

Reply:
You are right. Here's the additional info -
    _Header differences:
        \<Content-Length\> values differ, request <4358>, record <4274>_

I am recording again to see if the extra 84 bytes appear or if the recorded number stays the same.

Reply:
The issue is that the connection string used in the first (creation) call uses 'Sanitized' for the AccountKey part, but the service response includes the actual accesskey. Then, in the second (update) call, we uses the returned object and now the connection string has the real accesskey, causing the length difference.

Reply:
I think setting up a sanitizer using the JSON paths like the second example https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/core/Azure.Core.TestFramework#sanitizing is probably the way to go?

Reply:
That was quite a interesting deep dive into the world of recording session data and sanitizing.
After trying a few different approaches, I believe I have landed on a change with minimal changes needed on all sides.